web3.js/src/transaction.ts

```diff
@@ -231,6 +244,15 @@ export class Transaction {
    */
   nonceInfo?: NonceInformation;
 
+  /**
+   * If this is a nonce transaction this represents the minimum slot from which
+   * to evaluate if the nonce has advanced when attempting to confirm the
+   * transaction. This protects against a case where the transaction confirmation
+   * logic loads the nonce account from an old slot and assumes the mismatch in
+   * nonce value implies that the nonce has been advanced.
+   */
+  minNonceContextSlot?: number;
```

Comment:
I think this should be added to `NonceInformation` as an optional `contextSlot` field

Reply:
1. It shouldn’t be on the transaction at all, but we do so here to maintain compatibility with the send-and-confirm-transaction helper that does was too much work. That helper doesn’t accept this param, but rather expects it to live on the transaction object. That helper is literally the only reason that there exist `lastValidBlockHeight` and `minNonceContextSlot` properties on `Transaction`. Scorched earth.
2. Ooooh, no please not optional. We’re already drowning in optional values in this library, and it’s the cause of so many bugs. As you pointed out, the min slot is so important to make this thing work reliably, and I don’t want to undermine that by making it optional in the new strategy.

Reply:
> Couldn't those properties be passed to the helper then?

Absolutely, but then we'd have another multi-month method signature deprecation on our hands (ie. we'd have to turn over the method signature of `sendAndConfirmTransaction`)

> Why do either of them need to exist on the transaction object?

Because of the bad design of `sendAndConfirmTransaction` that only takes a `Transaction` as an argument.